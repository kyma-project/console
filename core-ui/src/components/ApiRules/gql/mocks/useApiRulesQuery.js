import { GET_API_RULES } from 'gql/queries';
import { API_RULE_EVENT_SUBSCRIPTION } from 'gql/subscriptions';
import { TESTING_STATE } from 'components/Lambdas/helpers/testing';

export const apiRuleMock = {
  name: 'apiRule',
  generation: 'generation',
  spec: {
    rules: [
      {
        path: '/.*',
        methods: ['GET', 'POST'],
        accessStrategies: [
          {
            handler: 'nope',
            config: {},
          },
        ],
      },
    ],
    service: {
      host: 'host',
      name: 'service',
      port: '80',
    },
  },
  status: {
    apiRuleStatus: {
      code: 'OK',
      description: '',
    },
  },
};

export const GET_API_RULES_ERROR_MOCK = variables => ({
  request: {
    query: GET_API_RULES,
    variables,
  },
  error: new Error(TESTING_STATE.ERROR),
});

export const GET_API_RULES_DATA_MOCK = (
  variables,
  APIRules = [apiRuleMock],
) => ({
  request: {
    query: GET_API_RULES,
    variables,
  },
  result: {
    data: {
      APIRules,
    },
  },
});

export const API_RULE_EVENT_SUBSCRIPTION_MOCK = (
  variables,
  apiRuleEvent = { type: 'ADD', apiRule: apiRuleMock },
) => ({
  request: {
    query: API_RULE_EVENT_SUBSCRIPTION,
    variables,
  },
  result: {
    data: {
      apiRuleEvent,
    },
  },
});
