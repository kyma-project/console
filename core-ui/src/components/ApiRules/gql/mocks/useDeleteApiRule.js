import { DELETE_API_RULE } from 'gql/mutations';
import { TESTING_STATE } from 'components/Lambdas/helpers/testing';

export const DELETE_API_RULE_ERROR_MOCK = variables => ({
  request: {
    query: DELETE_API_RULE,
    variables,
  },
  error: new Error(TESTING_STATE.ERROR),
});

export const DELETE_API_RULE_DATA_MOCK = variables => ({
  request: {
    query: DELETE_API_RULE,
    variables,
  },
  result: {
    data: {
      deleteAPIRule: {
        name: variables.name,
      },
    },
  },
});
