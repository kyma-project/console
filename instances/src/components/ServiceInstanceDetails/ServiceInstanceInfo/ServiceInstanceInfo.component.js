import React from 'react';
import Grid from 'styled-components-grid';

import {
  Icon,
  Header,
  Separator,
  InformationModal,
} from '@kyma-project/react-components';

import {
  ServiceInstanceInfoWrapper,
  StretchedContentWrapper,
  CenterSideWrapper,
  ContentHeader,
  ContentDescription,
  Element,
  InfoIcon,
  PlanModalButton,
  Label,
  ExternalLink,
  JSONCode,
} from './styled';

import { getResourceDisplayName, statusColor } from '../../../commons/helpers';

const ServiceInstanceInfo = ({ serviceInstance }) => {
  const statusIcon = statusType => {
    switch (statusType) {
      case ('PROVISIONING', 'DEPROVISIONING', 'PENDING'):
        return '\uE1C4';
      case 'FAILED':
        return '\uE0B1';
      case 'RUNNING':
        return '\uE1C2';
      default:
        return '\uE1C4';
    }
  };

  return (
    <ServiceInstanceInfoWrapper>
      <Grid.Unit size={0.7}>
        <CenterSideWrapper margin={'left'}>
          <StretchedContentWrapper>
            <ContentHeader>
              <Header>General Information</Header>
            </ContentHeader>
            <Separator />
            <ContentDescription>
              <Grid>
                <Grid.Unit size={0.2}>
                  <Element margin="0">Service Class</Element>
                </Grid.Unit>
                <Grid.Unit size={0.8}>
                  <Element margin="0" data-e2e-id="instance-service-class">
                    {serviceInstance.serviceClass
                      ? getResourceDisplayName(serviceInstance.serviceClass)
                      : '-'}
                  </Element>
                </Grid.Unit>
                <Grid.Unit size={0.2}>
                  <Element>Plan</Element>
                </Grid.Unit>
                <Grid.Unit size={0.8}>
                  <Element>
                    {serviceInstance.servicePlanSpec &&
                    Object.keys(serviceInstance.servicePlanSpec).length ? (
                      <InformationModal
                        modalOpeningComponent={
                          <PlanModalButton data-e2e-id="instance-service-plan">
                            {`${getResourceDisplayName(
                              serviceInstance.servicePlan,
                            )} plan`}
                          </PlanModalButton>
                        }
                        title="Instance Parameters"
                        content={
                          <JSONCode>
                            {JSON.stringify(
                              serviceInstance.servicePlanSpec,
                              undefined,
                              2,
                            )}
                          </JSONCode>
                        }
                      />
                    ) : (
                      `${getResourceDisplayName(
                        serviceInstance.servicePlan,
                      )} plan`
                    )}
                  </Element>
                </Grid.Unit>
              </Grid>
              {serviceInstance.labels &&
                serviceInstance.labels.length > 0 && (
                  <Grid>
                    <Grid.Unit size={0.2}>
                      <Element>Labels</Element>
                    </Grid.Unit>
                    <Grid.Unit size={0.8}>
                      <Element margin="1px 0 0 0">
                        {serviceInstance.labels.map(label => (
                          <Label key={label}>{label}</Label>
                        ))}
                      </Element>
                    </Grid.Unit>
                  </Grid>
                )}
              {serviceInstance.serviceClass &&
                (serviceInstance.serviceClass.documentationUrl ||
                  serviceInstance.serviceClass.supportUrl) && (
                  <Grid>
                    <Grid.Unit size={0.2}>
                      <Element margin="11px 0 0 0">Documentations</Element>
                    </Grid.Unit>
                    <Grid.Unit size={0.8}>
                      <Element margin="11px 0 0 0">
                        {serviceInstance.serviceClass.documentationUrl && (
                          <ExternalLink
                            href={serviceInstance.serviceClass.documentationUrl}
                            target="_blank"
                            data-e2e-id="instance-service-documentation-link"
                          >
                            Documentation
                          </ExternalLink>
                        )}
                        {serviceInstance.serviceClass.supportUrl ? ', ' : ''}
                        {serviceInstance.serviceClass.supportUrl && (
                          <ExternalLink
                            href={serviceInstance.serviceClass.supportUrl}
                            target="_blank"
                            data-e2e-id="instance-service-support-link"
                          >
                            Support
                          </ExternalLink>
                        )}
                      </Element>
                    </Grid.Unit>
                  </Grid>
                )}
            </ContentDescription>
          </StretchedContentWrapper>
        </CenterSideWrapper>
      </Grid.Unit>
      <Grid.Unit size={0.3}>
        <CenterSideWrapper>
          <StretchedContentWrapper
            color={statusColor(serviceInstance.status.type)}
          >
            <ContentHeader>
              <Grid>
                <Grid.Unit size={0.9}>
                  <Header>Status</Header>
                </Grid.Unit>
                <Grid.Unit size={0.1}>
                  <InfoIcon color={statusColor(serviceInstance.status.type)}>
                    <Icon icon={statusIcon(serviceInstance.status.type)} />
                  </InfoIcon>
                </Grid.Unit>
              </Grid>
            </ContentHeader>
            <Separator />
            <ContentDescription>
              <Element data-e2e-id="instance-status-type">
                {serviceInstance.status.type}
              </Element>
              <Element
                style={{ padding: '0' }}
                data-e2e-id="instance-status-message"
              >
                {serviceInstance.status.message}
              </Element>
            </ContentDescription>
          </StretchedContentWrapper>
        </CenterSideWrapper>
      </Grid.Unit>
    </ServiceInstanceInfoWrapper>
  );
};

export default ServiceInstanceInfo;
