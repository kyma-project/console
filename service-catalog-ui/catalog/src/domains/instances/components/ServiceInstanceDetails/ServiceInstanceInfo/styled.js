import styled from 'styled-components';
import { PanelGrid, Panel, PanelHeader } from '@kyma-project/react-components';

export const ServiceInstanceInfoWrapper = styled(PanelGrid)`
  margin: 0 32px;
  && > * {
    box-shadow: none;
  }
`;

export const StatusWrapper = styled(Panel)`
   {
    && {
      border: solid 1px #d9d9d9;
      border-left: ${props =>
        props.color ? '6px solid ' + props.color : 'none'};
    }
  }
`;

export const ContentHeader = styled(PanelHeader)`
  && {
    color: #32363a;
    font-size: 16px;
    font-weight: normal;
  }
`;

export const Element = styled.div`
  margin: ${props => (props.margin ? props.margin : '16px 0 0 0')};
`;

export const ServiceClassButton = styled.span`
  color: #0a6ed1;
  cursor: pointer;
`;

export const PlanModalButton = styled.span`
  font-size: 14px;
  font-weight: normal;
  display: inline-block;
  font-weight: 500;
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  background: transparent;
  color: #0b74de;
  cursor: pointer;
`;

export const LabelWrapper = styled.div`
  display: inline-block;
  margin: 10px 10px 0 0;
`;

export const ExternalLink = styled.a`
  && {
    font-size: 14px;
    color: #167ee6;
    font-weight: 500;
    text-decoration: none;

    :hover {
      text-decoration: underline;
    }
  }
`;

export const JSONCode = styled.code`
  max-width: 70vw;
  display: inline-block;
  white-space: pre-wrap;
  white-space: -moz-pre-wrap;
  white-space: -o-pre-wrap;
  word-wrap: break-word;
`;

export const DescriptionKey = styled.p`
  margin-bottom: 0;
  margin-top: 24px;
  color: #6a6d70;
  font-size: 14px;
`;
