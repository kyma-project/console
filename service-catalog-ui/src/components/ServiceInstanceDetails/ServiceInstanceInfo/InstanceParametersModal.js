import React from 'react';
import PropTypes from 'prop-types';

import { Modal, JSONEditor } from 'react-shared';
import { getResourceDisplayName } from 'helpers';
import { serviceInstanceConstants } from 'helpers/constants';

InstanceParametersModal.propTypes = {
  serviceInstance: PropTypes.object.isRequired,
  instancePlan: PropTypes.object.isRequired,
};

export default function InstanceParametersModal({
  serviceInstance,
  instancePlan,
}) {
  const formattedPlan = JSON.stringify(serviceInstance.planSpec, null, 2);

  const modalOpeningComponent = (
    <button className="link has-padding-0" option="light">
      {getResourceDisplayName(instancePlan)}
    </button>
  );

  return (
    <Modal
      modalOpeningComponent={modalOpeningComponent}
      title={serviceInstanceConstants.instanceParameters}
      confirmText="Close"
    >
      <JSONEditor readonly={true} text={formattedPlan} />
    </Modal>
  );
}
