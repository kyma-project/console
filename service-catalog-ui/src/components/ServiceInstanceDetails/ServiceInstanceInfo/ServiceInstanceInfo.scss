.labels-list > * {
  margin: 10px 10px 0 0 !important;
}
