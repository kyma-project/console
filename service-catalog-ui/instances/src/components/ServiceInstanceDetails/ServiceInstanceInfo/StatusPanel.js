import React from 'react';
import { StatusWrapper, ContentHeader, Element } from './styled';
import {
  Icon,
  instanceStatusColor,
  PanelBody,
  PanelActions,
} from '@kyma-project/react-components';
import { serviceInstanceConstants } from '../../../variables';

const statusIcon = statusType => {
  switch (statusType) {
    case 'FAILED':
      return 'sys-cancel';
    case 'RUNNING':
      return 'sys-enter';
    default:
      return 'sys-help';
  }
};

export const StatusPanel = ({ serviceInstance }) => {
  return (
    <StatusWrapper
      data-e2e-id="instance-status"
      colSpan={1}
      color={instanceStatusColor(serviceInstance.status.type)}
    >
      <ContentHeader>
        {serviceInstanceConstants.statusHeader}
        <PanelActions>
          <Icon
            glyph={statusIcon(serviceInstance.status.type)}
            style={{
              color: instanceStatusColor(serviceInstance.status.type),
            }}
          />
        </PanelActions>
      </ContentHeader>

      <PanelBody>
        <Element
          margin="0"
          style={{ color: instanceStatusColor(serviceInstance.status.type) }}
        >
          {serviceInstance.status.type}
        </Element>
        <Element style={{ padding: '0' }}>
          {serviceInstance.status.message}
        </Element>
      </PanelBody>
    </StatusWrapper>
  );
};
