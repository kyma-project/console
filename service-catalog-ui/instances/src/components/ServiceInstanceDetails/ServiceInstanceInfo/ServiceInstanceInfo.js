import React from 'react';
import LuigiClient from '@kyma-project/luigi-client';
import Grid from 'styled-components-grid';
import { getResourceDisplayName } from '../../../commons/helpers';
import { serviceInstanceConstants } from '../../../variables';

import { Modal, PanelBody, Label, Panel } from '@kyma-project/react-components';

import {
  ServiceInstanceInfoWrapper,
  Element,
  PlanModalButton,
  ServiceClassButton,
  ExternalLink,
  JSONCode,
  DescriptionKey,
  LabelWrapper,
} from './styled';
import { StatusPanel } from './StatusPanel';

const INFORMATION_CELL_SIZE = { mobile: 1, tablet: 0.5, desktop: 0.5 };

const ServiceInstanceInfo = ({ serviceInstance }) => {
  const goToServiceClassDetails = name => {
    LuigiClient.linkManager()
      .fromContext('namespaces')
      .navigate(`cmf-service-catalog/details/${name}`);
  };

  if (!serviceInstance) {
    return null;
  }

  const instanceClass = serviceInstance.clusterServiceClass
    ? serviceInstance.clusterServiceClass
    : serviceInstance.serviceClass;
  const instancePlan = serviceInstance.clusterServicePlan
    ? serviceInstance.clusterServicePlan
    : serviceInstance.servicePlan;

  return (
    <ServiceInstanceInfoWrapper
      cols={3}
      className="fd-has-padding-bottom-medium"
    >
      <Panel colSpan={2}>
        <PanelBody className="fd-has-margin-bottom-medium fd-has-padding-none">
          <Grid>
            <Grid.Unit size={INFORMATION_CELL_SIZE}>
              <DescriptionKey>
                {serviceInstanceConstants.classHeader}
              </DescriptionKey>
              <Element margin="0" data-e2e-id="instance-service-class">
                {instanceClass && instanceClass.name ? (
                  <ServiceClassButton
                    onClick={() => goToServiceClassDetails(instanceClass.name)}
                  >
                    {getResourceDisplayName(instanceClass)}
                  </ServiceClassButton>
                ) : (
                  '-'
                )}
              </Element>
            </Grid.Unit>
            <Grid.Unit size={INFORMATION_CELL_SIZE}>
              <DescriptionKey>
                {serviceInstanceConstants.planHeader}
              </DescriptionKey>
              <Element margin="0" data-e2e-id="instance-service-plan">
                {serviceInstance.planSpec &&
                serviceInstance.planSpec !== null &&
                typeof serviceInstance.planSpec === 'object' &&
                Object.keys(serviceInstance.planSpec).length ? (
                  <Modal
                    modalOpeningComponent={
                      <PlanModalButton>
                        {getResourceDisplayName(instancePlan)}
                      </PlanModalButton>
                    }
                    title={serviceInstanceConstants.instanceParameters}
                    onShow={() => LuigiClient.uxManager().addBackdrop()}
                    onHide={() => LuigiClient.uxManager().removeBackdrop()}
                  >
                    <JSONCode>
                      {JSON.stringify(serviceInstance.planSpec, undefined, 2)}
                    </JSONCode>
                  </Modal>
                ) : (
                  `${getResourceDisplayName(instancePlan) || '-'}`
                )}
              </Element>
            </Grid.Unit>
          </Grid>
          <Grid>
            {instanceClass && instanceClass.documentationUrl ? (
              <Grid.Unit size={INFORMATION_CELL_SIZE}>
                <DescriptionKey>
                  {serviceInstanceConstants.documentationHeader}
                </DescriptionKey>
                <Element
                  margin="0"
                  data-e2e-id="instance-service-documentation-link"
                >
                  <ExternalLink
                    href={instanceClass.documentationUrl}
                    target="_blank"
                  >
                    {serviceInstanceConstants.link}
                  </ExternalLink>
                </Element>
              </Grid.Unit>
            ) : null}

            {instanceClass && instanceClass.supportUrl ? (
              <Grid.Unit size={INFORMATION_CELL_SIZE}>
                <DescriptionKey>
                  {serviceInstanceConstants.supportHeader}
                </DescriptionKey>
                <Element margin="0" data-e2e-id="instance-service-support-link">
                  <ExternalLink href={instanceClass.supportUrl} target="_blank">
                    {serviceInstanceConstants.link}
                  </ExternalLink>
                </Element>
              </Grid.Unit>
            ) : null}
          </Grid>
          <Grid>
            {serviceInstance.labels && serviceInstance.labels.length > 0 && (
              <div>
                <DescriptionKey>
                  {serviceInstanceConstants.labelsHeader}
                </DescriptionKey>
                <Element margin="0" data-e2e-id="instance-label">
                  {serviceInstance.labels.map((label, index) => (
                    <LabelWrapper key={`${label}-${index}`}>
                      <Label key={label} cursorType="auto">
                        {label}
                      </Label>
                    </LabelWrapper>
                  ))}
                </Element>
              </div>
            )}
          </Grid>
        </PanelBody>
      </Panel>
      <StatusPanel serviceInstance={serviceInstance} />
    </ServiceInstanceInfoWrapper>
  );
};

export default ServiceInstanceInfo;
