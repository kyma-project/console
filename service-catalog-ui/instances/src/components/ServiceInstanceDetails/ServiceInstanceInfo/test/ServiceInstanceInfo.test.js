import { mount } from 'enzyme';
import ServiceInstanceInfo from '../ServiceInstanceInfo';
import React from 'react';
import { serviceInstance1 } from './mocks';

describe('ServiceInstanceInfo', () => {
  it('Render info with all attributes', () => {
    const component = mount(
      <ServiceInstanceInfo serviceInstance={serviceInstance1} />,
    );
  });
  it('Render info without labels', () => {});
  it('Render info without support', () => {});
  it('Render info without documentation link', () => {});
  it('Render info without plan parameters', () => {});

  [
    { status: 'PROVISIONING', class: 'sys-help' },
    { status: 'FAILED', class: 'sys-cancel' },
    { status: 'RUNNING', class: 'sys-enter' },
  ].map(testCase => {
    it(`Render info with status ${testCase.status}`, () => {});
  });
});
