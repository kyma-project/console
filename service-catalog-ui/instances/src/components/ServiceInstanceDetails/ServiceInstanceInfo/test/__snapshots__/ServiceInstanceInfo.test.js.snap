// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceInstanceInfo Render info with status FAILED 1`] = `ReactWrapper {}`;

exports[`ServiceInstanceInfo Render info with status PROVISIONING 1`] = `ReactWrapper {}`;

exports[`ServiceInstanceInfo Render info with status RUNNING 1`] = `ReactWrapper {}`;
